### PR TITLE
Feature/3.2 rest protocol provider

### PR DESCRIPTION
**Pull request by suncairong163**
## What is the purpose of the change

RestProtocol provider  restHandler & restProtocol 

## Brief changelog
##Related issues

#10803 
@AlbumenJ

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

File: dubbo-remoting/dubbo-remoting-http/src/main/java/org/apache/dubbo/remoting/http/HttpHandler.java [hunk 1]
```diff
@@ -34,6 +32,6 @@ public interface HttpHandler {
      * @throws IOException
      * @throws ServletException
      */
-    void handle(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException;
+    void handle(REQUEST request, RESPONSE response) throws IOException, ServletException;
```
Comment (icodening) [inline dubbo-remoting/dubbo-remoting-http/src/main/java/org/apache/dubbo/remoting/http/HttpHandler.java:35]: will throw an `java.lang.NoClassDefFoundError` if the classpath does not have `ServletException`

File: dubbo-rpc/dubbo-rpc-rest/src/main/java/org/apache/dubbo/rpc/protocol/rest/request/RequestFacadeFactory.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        try {
+            Class<?> requestClass = classLoader.loadClass(requestClassName);
+
+            return requestClass.isAssignableFrom(request.getClass());
+
+        } catch (ClassNotFoundException e) {
+            return false;
+        }
+
+    }
```
Comment (icodening) [inline dubbo-rpc/dubbo-rpc-rest/src/main/java/org/apache/dubbo/rpc/protocol/rest/request/RequestFacadeFactory.java:63]: can be replace `org.apache.dubbo.common.utils.ClassUtils#isPresent`

File: dubbo-rpc/dubbo-rpc-rest/src/main/java/org/apache/dubbo/rpc/protocol/rest/handler/NettyHttpHandler.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+//            dispatcher.service(request, servletResponse);
+
+        Pair<RpcInvocation, Invoker> build = null;
+        try {
+            build = RPCInvocationBuilder.build(request, nettyHttpRequest, nettyHttpResponse);
+        } catch (PathNoFoundException e) {
+            nettyHttpResponse.setStatus(404);
+        }
+
+        Invoker invoker = build.getSecond();
```
Comment (icodening) [inline dubbo-rpc/dubbo-rpc-rest/src/main/java/org/apache/dubbo/rpc/protocol/rest/handler/NettyHttpHandler.java:56]: will throw NPE if  invoke `RPCInvocationBuilder.build` error

File: dubbo-rpc/dubbo-rpc-rest/src/main/java/org/apache/dubbo/rpc/protocol/rest/DubboHttpProtocolServer.java [hunk 4]
```diff
@@ -80,12 +93,64 @@ protected ResteasyDeployment getDeployment() {
...
+//            dispatcher.service(request, servletResponse);
+
+            Pair<RpcInvocation, Invoker> build = null;
+            try {
+                build = RPCInvocationBuilder.build(request, servletRequest, servletResponse);
+            } catch (PathNoFoundException e) {
+                servletResponse.setStatus(404);
+            }
+
+            Invoker invoker = build.getSecond();
```
Comment (icodening) [inline dubbo-rpc/dubbo-rpc-rest/src/main/java/org/apache/dubbo/rpc/protocol/rest/DubboHttpProtocolServer.java:112]: will throw NPE if invoke `RPCInvocationBuilder.build` error

File: dubbo-rpc/dubbo-rpc-rest/src/main/java/org/apache/dubbo/rpc/protocol/rest/PathAndInvokerMapper.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+import org.apache.dubbo.metadata.rest.PathMatcher;
+import org.apache.dubbo.metadata.rest.RestMethodMetadata;
+import org.apache.dubbo.rpc.Invoker;
+import org.apache.dubbo.rpc.protocol.rest.exception.PathNoFoundException;
+import org.apache.dubbo.rpc.protocol.rest.util.Pair;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class PathAndInvokerMapper {
```
Comment (icodening) [inline dubbo-rpc/dubbo-rpc-rest/src/main/java/org/apache/dubbo/rpc/protocol/rest/PathAndInvokerMapper.java:28]: It seems to be similar to `org.apache.dubbo.rpc.PathResolver`?

**Review by icodening — CHANGES_REQUESTED** [review]

File: dubbo-rpc/dubbo-rpc-rest/src/main/java/org/apache/dubbo/rpc/protocol/rest/PathAndInvokerMapper.java (hunk 5 above)
Comment (suncairong163) [reply dubbo-rpc/dubbo-rpc-rest/src/main/java/org/apache/dubbo/rpc/protocol/rest/PathAndInvokerMapper.java:28]: 是的，这块功能上是和pathResolver一样的，但是会有些不一样的，org.apache.dubbo.rpc.PathResolver#add(String path, Invoker<?> invoker)    这块需要抽象成泛型，

File: dubbo-remoting/dubbo-remoting-http/src/main/java/org/apache/dubbo/remoting/http/HttpHandler.java (hunk 1 above)
Comment (suncairong163) [reply dubbo-remoting/dubbo-remoting-http/src/main/java/org/apache/dubbo/remoting/http/HttpHandler.java:35]: ok,i will fix

File: dubbo-rpc/dubbo-rpc-rest/src/main/java/org/apache/dubbo/rpc/protocol/rest/request/RequestFacadeFactory.java (hunk 2 above)
Comment (suncairong163) [reply dubbo-rpc/dubbo-rpc-rest/src/main/java/org/apache/dubbo/rpc/protocol/rest/request/RequestFacadeFactory.java:63]: ok，I will fix

File: dubbo-rpc/dubbo-rpc-rest/src/main/java/org/apache/dubbo/rpc/protocol/rest/handler/NettyHttpHandler.java (hunk 3 above)
Comment (suncairong163) [reply dubbo-rpc/dubbo-rpc-rest/src/main/java/org/apache/dubbo/rpc/protocol/rest/handler/NettyHttpHandler.java:56]: ok，I will fix

File: dubbo-rpc/dubbo-rpc-rest/src/main/java/org/apache/dubbo/rpc/protocol/rest/DubboHttpProtocolServer.java (hunk 4 above)
Comment (suncairong163) [reply dubbo-rpc/dubbo-rpc-rest/src/main/java/org/apache/dubbo/rpc/protocol/rest/DubboHttpProtocolServer.java:112]: ok，I will fix

**Review by AlbumenJ — APPROVED** [review]
LGTM

**Pull request merged**